### DISCUSSION TITLE
Multifile Indices

### DISCUSSION THREAD

**Discussion by naturfenomen** [visose/Robots#72] [Q&A] [answered]
Hi,

How do I use the 'Multifile indices' parameter in the 'Create program' node? - I'm trying to split my file as our UR10e struggles with larger files. And I cannot find any examples of this parameter being used.

**Comment by visose** [comment, marked as answer]
I added the Discussions section 5 minutes ago, you were fast 😄

Multi file only works in ABB, KUKA and Staubli. Not in UR, sorry. A workaround you can do for now is split the program in Grasshopper using branches or similar then select each branch individually running them one by one manually.

I'll add a warning about multi-file not working with UR.

**Reply by naturfenomen** [reply]
Thank you for the fast answer - it was driving me nuts!

And I didn't know you just added the discussions - But accidentally fast is still fast!

**Comment by PhantomStaubli** [comment]
can you explain how you are using the Multifile Indices with an example?

**Reply by visose** [reply]
You just need to connect a list of integers. It will create a separate code file per integer, where each integer corresponds to the index of the target that each file will begin with. If you don't include 0, it will be implicitly added.

**Reply by PhantomStaubli** [reply]
I tried to break the .dtx file which is over 20MB and it doesn't work. I was able to break .pgx into multiple files but the controller still freezes. I would really appreciate an example for Staubli robots. I read the entire Val3 manual and nothing works. Thanks again

**Reply by visose** [reply]
Hi, I never tested Staubli with a large program that runs out of memory. The current implementation only splits the pgx files but not the dtx file. It's possible that the dtx file also needs to be split if it becomes too large.
I'm not sure when I might have a chance to get to implement and test this. If you are willing to sponsor this feature I can prioritize it.
